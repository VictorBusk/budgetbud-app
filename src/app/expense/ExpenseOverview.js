import React, {Component} from 'react';
import {
    Body,
    Container,
    Icon,
    ListItem,
    Text,
    View,
    Right,
    Spinner
} from "native-base";
import {AppHeader, Separator} from "../../components";
import {FlatList, StyleSheet, TouchableOpacity} from "react-native";
import {connect} from "react-redux";
import I18n from "../../strings/i18n";
import {color, container} from "../../style";
import {
    getCategoryAlarms,
    toggleCategoryAlarm,
    getExpensesOfMonth
} from "../../redux/actions";

class ExpenseOverview extends Component {
    async componentWillMount() {
        if (!this.props.categoryAlarmsInitialized)
            this.props.getCategoryAlarms(this.props.budgetID);

        if (!this.props.expensesInitialized)
            this.props.getExpensesOfMonth();
    }

    onAlarmPress = async ({categoryID}) => {
        if (!this.props.toggleLoading)
            this.props.toggleCategoryAlarm(categoryID, this.props.budgetID)
    };

    render() {
        return (
            <Container style={container.signedInContainer}>
                <AppHeader headerText={I18n.t('expenseOverviewHeader')}
                           onLeftButtonPress={() => this.props.navigation.navigate("DrawerOpen")}/>
                <Container>
                    {this.props.expensesLoading ? (
                        <Spinner style={{
                            flex: 1
                        }} color='#1c313a'/>) : (
                        <Container>
                            <View style={{flex: 0.82}}>
                                <FlatList
                                    data={this.props.categoryItems}
                                    renderItem={this.renderCategory}
                                    keyExtractor={item => item.categoryID}
                                />
                                <Separator/>
                            </View>
                            <View style={{flex: 0.18}}>
                                <View style={[styles.budgetSummary, {paddingVertical: '5%',}]}>
                                    <View style={[styles.incomeFormStyle, {flex: 1}]}>
                                        <Text style={styles.listText}>
                                            {I18n.t('expenseOverviewTotalDebtsPerMonth')}
                                        </Text>
                                        <Text style={styles.listText}>
                                            {this.props.totalDebtPerMonth} {I18n.t('currency')}
                                        </Text>
                                    </View>
                                    <View style={[styles.incomeFormStyle, {flex: 1}]}>
                                        <Text style={styles.listText}>
                                            {I18n.t('expenseOverviewTotalExpenses')}
                                        </Text>
                                        <Text style={styles.listText}>
                                            {this.props.totalExpenses} {I18n.t('currency')}
                                        </Text>
                                    </View>
                                    <View style={[styles.incomeFormStyle, {flex: 1}]}>
                                        <Text style={styles.listText}>
                                            {I18n.t('expenseOverviewDisposable')}
                                        </Text>
                                        <Text style={styles.listText}>
                                            {this.props.disposable} {I18n.t('currency')}
                                        </Text>
                                    </View>
                                </View>
                            </View>
                        </Container>
                    )}
                </Container>
            </Container>
        );
    }

    renderCategory = ({item}) => {
        const notificationsEnabled = this.props.categoryAlarms.includes(item.categoryID);

        return (
            <ListItem>
                <View style={{width: '100%', paddingHorizontal: 4}}>
                    <View style={[container.removeIndenting, {flexDirection: 'row'}]}>
                        <Body style={{alignItems: 'flex-start'}}>
                        <Text style={color.text}>{item.name}</Text>
                        </Body>
                        <Right>
                            <TouchableOpacity onPress={() => this.onAlarmPress(item)}>
                                <Icon style={[color.darkIcon, {fontSize: 26}]}
                                      name={notificationsEnabled ? "ios-notifications" : "ios-notifications-outline"}/>
                            </TouchableOpacity>
                        </Right>
                    </View>
                    <View style={[container.removeIndenting, {flexDirection: 'row'}]}>
                        <Body style={{alignItems: 'flex-start'}}>
                        <Text style={{color: '#808080'}}>{I18n.t('expenseOverviewBudget')}</Text>
                        <Text style={{color: '#808080'}}>{I18n.t('expenseOverviewExpenses')}</Text>
                        <Text style={{color: '#808080'}}>{I18n.t('expenseOverviewRemaining')}</Text>
                        </Body>
                        <Right>
                            <Text style={{color: '#808080'}}>{item.budget} {I18n.t('currency')}</Text>
                            <Text style={{color: '#808080'}}>{item.expenses} {I18n.t('currency')}</Text>
                            <Text
                                style={((item.budget - item.expenses) < 0 ? {color: '#EB7C83'} :
                                    {color: '#808080'})}>{item.budget - item.expenses} {I18n.t('currency')}
                            </Text>
                        </Right>
                    </View>
                </View>
            </ListItem>
        );
    };
}

const styles = StyleSheet.create({
    buttonStyle: {
        justifyContent: 'flex-end',
        alignSelf: 'flex-end',
        paddingBottom: 10
    },
    incomeFormStyle: {
        alignSelf: 'center',
        width: '100%',
        paddingRight: 18,
        paddingLeft: 15,
        justifyContent: 'space-between',
        flexDirection: 'row'
    },
    expenseListStyle: {
        alignSelf: 'center',
        width: '90%',
        justifyContent: 'space-between',
        flexDirection: 'row',
        paddingTop: 23
    },
    itemStyle: {
        fontWeight: '600',
        alignSelf: 'flex-start',
        color: 'white'
    },
    spacedText: {
        flexGrow: 1,
        flexDirection: 'row',
        justifyContent: 'space-between',
        alignItems: 'center',
        marginHorizontal: '5%',
        marginVertical: 10
    },
    budgetSummary: {
        marginBottom: 0,
        justifyContent: 'space-between',
        flex: 1
    },
    textStyle: {
        fontWeight: '400',
        fontSize: 14,
        alignSelf: 'flex-start',
        marginLeft: 5,
        color: '#00263A'
    },
    listText: {
        marginLeft: 5,
        alignSelf: 'flex-start',
        fontSize: 16,
        color: '#00263A'
    },
    modalButton: {
        flexDirection: 'column',
        justifyContent: 'center',
        alignItems: 'center'
    }
});

const mapStateToProps = (state) => {
    const {budgetID} = state.budget;
    const {debts} = state.debt;
    const {expenses, totalExpenses, expensesLoading, expensesInitialized} = state.expense;
    const disposable = state.disposable.disposable;
    const {categories} = state.category;
    const {categoryAlarms, toggleLoading, categoryAlarmsInitialized} = state.alarm;

    let totalDebtPerMonth = 0;
    debts.forEach(d => totalDebtPerMonth += d.debtData.amountPerMonth);

    const categoryItems = [];

    categories.forEach(category => {
        const index = expenses.findIndex(x => x.categoryTypeID === category.categoryTypeID);

        if (index !== -1) {
            categoryItems.push({
                categoryID: category.categoryID,
                name: category.name,
                budget: category.amount,
                expenses: expenses[index].amount,
                notificationsEnabled: false
            });
        }
    });

    return {
        budgetID,
        categoryAlarms,
        toggleLoading,
        totalDebtPerMonth,
        disposable,
        expenses,
        totalExpenses,
        expensesLoading,
        categoryItems,
        expensesInitialized,
        categoryAlarmsInitialized
    }
};

export default connect(mapStateToProps, {
    getExpensesOfMonth, getCategoryAlarms, toggleCategoryAlarm
})(ExpenseOverview);
