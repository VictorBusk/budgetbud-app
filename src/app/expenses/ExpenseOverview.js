import React, {Component} from 'react';
import {
    Body,
    Container,
    Icon,
    ListItem,
    Text,
    View,
    Right,
    Spinner
} from "native-base";
import {AppHeader, Separator} from "../../components";
import {FlatList, TouchableOpacity} from "react-native";
import {connect} from "react-redux";
import I18n from "../../strings/i18n";
import {color, container, text} from "../../style";
import {
    getCategoryAlarms,
    toggleCategoryAlarm,
    getExpensesOfMonth,
    resetExpensesError
} from "../../redux/actions";
import {showWarningToast} from "../../helpers/toasts";

class ExpenseOverview extends Component {
    componentWillMount() {
        if (!this.props.categoryAlarmsInitialized)
            this.props.getCategoryAlarms(this.props.budgetID);

        if (!this.props.expensesInitialized)
            this.props.getExpensesOfMonth();
    }

    componentWillReceiveProps(nextProps) {
        if (nextProps.expensesError) {
            showWarningToast(nextProps.expensesError);
            this.props.resetExpensesError();
        }
    }

    onAlarmPress = async ({categoryID}) => {
        this.props.toggleCategoryAlarm(categoryID, this.props.budgetID)
    };

    render() {
        return (
            <Container style={container.signedInContainer}>
                <AppHeader headerText={I18n.t('expenseOverviewHeader')}
                           infoText={I18n.t('expenseOverviewInfo')}
                           onLeftButtonPress={() => this.props.navigation.navigate("DrawerOpen")}/>
                <Container>
                    {((this.props.expensesLoading || this.props.alarmsLoading)
                        && (!this.props.expensesInitialized && !this.props.categoriesInitialized)) ? (
                        <Spinner style={{
                            flex: 1
                        }} color='#1c313a'/>) : (
                        <Container>
                            <View style={{flex: 0.82}}>
                                <FlatList
                                    data={this.props.categoryItems}
                                    renderItem={this.renderCategory}
                                    keyExtractor={item => item.categoryID}
                                    refreshing={this.props.expensesLoading}
                                    onRefresh={() => this.props.getExpensesOfMonth()}
                                />
                                <Separator/>
                            </View>
                            <View style={{flex: 0.18}}>
                                <View style={[container.budgetSummary, {paddingVertical: '5%',}]}>
                                    <View style={[container.amountSummaryContainer, {flex: 1}]}>
                                        <Text style={[text.listText, color.text]}>
                                            {I18n.t('expenseOverviewTotalDebtsPerMonth')}
                                        </Text>
                                        <Text style={[text.listText, color.text]}>
                                            {this.props.totalDebtPerMonth} {I18n.t('currency')}
                                        </Text>
                                    </View>
                                    <View style={[container.amountSummaryContainer, {flex: 1}]}>
                                        <Text style={[text.listText, color.text]}>
                                            {I18n.t('expenseOverviewTotalExpenses')}
                                        </Text>
                                        <Text style={[text.listText, color.text]}>
                                            {this.props.totalExpenses} {I18n.t('currency')}
                                        </Text>
                                    </View>
                                    <View style={[container.amountSummaryContainer, {flex: 1}]}>
                                        <Text style={[text.listText, color.text]}>
                                            {I18n.t('expenseOverviewDisposable')}
                                        </Text>
                                        <Text style={[text.listText, color.text]}>
                                            {this.props.disposable} {I18n.t('currency')}
                                        </Text>
                                    </View>
                                </View>
                            </View>
                        </Container>
                    )}
                </Container>
            </Container>
        );
    }

    renderCategory = ({item}) => {
        const notificationsEnabled = this.props.categoryAlarms.includes(item.categoryID);

        return (
            <ListItem>
                <View style={{width: '100%', paddingHorizontal: 4}}>
                    <View style={[container.removeIndenting, {flexDirection: 'row'}]}>
                        <Body style={{alignItems: 'flex-start'}}>
                        <Text style={color.text}>{item.name}</Text>
                        </Body>
                        <Right>
                            <TouchableOpacity onPress={() => {
                                if (!this.props.toggleLoading)
                                    this.onAlarmPress(item)
                            }}>
                                <Icon style={[color.darkIcon, {fontSize: 26}]}
                                      name={notificationsEnabled ? "ios-notifications" : "ios-notifications-outline"}/>
                            </TouchableOpacity>
                        </Right>
                    </View>
                    <View style={[container.removeIndenting, {flexDirection: 'row'}]}>
                        <Body style={{alignItems: 'flex-start'}}>
                        <Text style={{color: '#808080'}}>{I18n.t('expenseOverviewBudget')}</Text>
                        <Text style={{color: '#808080'}}>{I18n.t('expenseOverviewExpenses')}</Text>
                        <Text style={{color: '#808080'}}>{I18n.t('expenseOverviewRemaining')}</Text>
                        </Body>
                        <Right>
                            <Text style={{color: '#808080'}}>{item.budget} {I18n.t('currency')}</Text>
                            <Text style={{color: '#808080'}}>{item.expenses} {I18n.t('currency')}</Text>
                            <Text
                                style={((item.budget - item.expenses) < 0 ? {color: '#EB7C83'} :
                                    {color: '#808080'})}>{item.budget - item.expenses} {I18n.t('currency')}
                            </Text>
                        </Right>
                    </View>
                </View>
            </ListItem>
        );
    };
}

const mapStateToProps = (state) => {
    const {budgetID} = state.budget;
    const {debts} = state.debt;
    const {
        expenses,
        totalExpenses,
        expensesLoading,
        expensesInitialized,
        expensesError
    } = state.expense;
    const disposable = state.disposable.disposable;
    const {categories} = state.category;
    const {categoryAlarms, toggleLoading, alarmsLoading, categoryAlarmsInitialized} = state.alarm;

    let totalDebtPerMonth = 0;
    debts.forEach(d => totalDebtPerMonth += d.debtData.amountPerMonth);

    const categoryItems = [];

    categories.forEach(category => {
        const index = expenses.findIndex(x => x.categoryTypeID === category.categoryTypeID);

        if (index !== -1) {
            categoryItems.push({
                categoryID: category.categoryID,
                name: category.name,
                budget: category.amount,
                expenses: expenses[index].amount,
                notificationsEnabled: false
            });
        }
    });

    return {
        budgetID,
        categoryAlarms,
        toggleLoading,
        totalDebtPerMonth,
        disposable,
        expenses,
        totalExpenses,
        expensesLoading,
        categoryItems,
        expensesInitialized,
        categoryAlarmsInitialized,
        expensesError,
        alarmsLoading
    }
};

export default connect(mapStateToProps, {
    getExpensesOfMonth,
    getCategoryAlarms,
    toggleCategoryAlarm,
    resetExpensesError
})(ExpenseOverview);
