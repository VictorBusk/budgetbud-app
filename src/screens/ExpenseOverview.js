import React, {Component} from 'react';
import {
    Body,
    Button,
    Container,
    Icon,
    ListItem,
    Text,
    View,
    Label,
    Spinner
} from "native-base";
import {AppHeader, Separator} from "../components/";
import {FlatList, StyleSheet} from "react-native";
import {connect} from "react-redux";
import {getBudget, getCategories, getDebts} from "../actions";
import I18n from "../strings/i18n";
import {container} from "../style";

class ExpenseOverview extends Component {
    componentWillMount() {
        this.props.getBudget(this.props.budgetID);
        this.props.getCategories(this.props.budgetID);
    }

    navigateUser = (destination) => {
        this.props.navigation.navigate(destination)
    };

    render() {
        return (
            <Container style={container.signedInContainer}>
                <AppHeader headerText={I18n.t('expenseOverviewHeader')}
                           onLeftButtonPress={() => this.props.navigation.navigate("DrawerOpen")}/>
                <Container>
                    {this.props.budgetLoading ? (
                        <Spinner style={{
                            flex: 1
                        }} color='#1c313a'/>) : (
                        <Container>
                            <View style={{flex: 0.82, paddingTop: 20}}>
                                    <View style={[styles.incomeFormStyle,]}>
                                        <Label style={styles.textStyle}>{I18n.t('budgetExpenses')}
                                        </Label>
                                    </View>
                                    <FlatList
                                        data={this.props.categories}
                                        renderItem={this.renderCategory}
                                        keyExtractor={item => item.id}
                                    />
                                <Separator/>
                            </View>

                            <View style={{flex: 0.18}}>
                                <View style={[styles.budgetSummary, {paddingVertical: '5%', }]}>
                                    <View style={[styles.incomeFormStyle, {flex: 1}]}>
                                        <Text style={styles.listText}>
                                            {I18n.t('budgetTotalExpenses')}
                                        </Text>
                                        <Text style={styles.listText}>
                                            {this.props.totalExpenses} {I18n.t('currency')}
                                        </Text>
                                    </View>
                                    <View style={[styles.incomeFormStyle, {flex: 1}]}>
                                        <Text style={styles.listText}>
                                            {I18n.t('budgetDisposable')}
                                        </Text>
                                        <Text style={styles.listText}>
                                            {this.props.disposable} {I18n.t('currency')}
                                        </Text>
                                    </View>
                                    <View style={[styles.incomeFormStyle, {flex: 1}]}>
                                        <Text style={styles.listText}>
                                            {I18n.t('budgetDisposable')}
                                        </Text>
                                        <Text style={styles.listText}>
                                            {this.props.disposable} {I18n.t('currency')}
                                        </Text>
                                    </View>
                                </View>
                            </View>
                        </Container>
                    )}
                </Container>
            </Container>
        );
    }

    renderCategory = ({item}) => {
        return (
            <ListItem style={{paddingLeft: 6, paddingRight: 18}}>
                <Body>
                <Text style={styles.listText}>{item.categoryData.name}</Text>
                </Body>
                <Text style={[styles.listText, {justifyContent: 'flex-end'}]}>
                    {item.categoryData.amount} {I18n.t('currency')}
                </Text>
            </ListItem>
        );
    };
}

const styles = StyleSheet.create({
    buttonStyle: {
        justifyContent: 'flex-end',
        alignSelf: 'flex-end',
        paddingBottom: 10
    },
    incomeFormStyle: {
        alignSelf: 'center',
        width: '90%',
        justifyContent: 'space-between',
        flexDirection: 'row'
    },
    expenseListStyle: {
        alignSelf: 'center',
        width: '90%',
        justifyContent: 'space-between',
        flexDirection: 'row',
        paddingTop: 23
    },
    itemStyle: {
        fontWeight: '600',
        alignSelf: 'flex-start',
        color: 'white'
    },
    spacedText: {
        flexGrow: 1,
        flexDirection: 'row',
        justifyContent: 'space-between',
        alignItems: 'center',
        marginHorizontal: '5%',
        marginVertical: 10
    },
    budgetSummary: {
        marginBottom: 0,
        justifyContent: 'space-between',
        flex: 1
    },
    textStyle: {
        fontWeight: '400',
        fontSize: 14,
        alignSelf: 'flex-start',
        marginLeft: 5,
    },
    listText: {
        marginLeft: 5,
        alignSelf: 'flex-start',
        fontSize: 16
    },
    modalButton: {
        flexDirection: 'column',
        justifyContent: 'center',
        alignItems: 'center'
    }
});

const mapStateToProps = (state) => {
    const categories = state.category.categories;
    const {
        budgetLoading,
        income,
        totalExpenses,
        disposable,
        destination,
        budgetID
    } = state.budget;

    return {
        categories,
        budgetLoading,
        income,
        totalExpenses,
        disposable,
        destination,
        budgetID
    }
};

export default connect(mapStateToProps, {
    getBudget, getCategories, getDebts
})(ExpenseOverview);
