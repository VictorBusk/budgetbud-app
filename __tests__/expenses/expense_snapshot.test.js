import React from 'react';
import configureStore from 'redux-mock-store';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import thunk from 'redux-thunk';
import ExpenseOverview from "../../src/app/expenses/ExpenseOverview";
import {
    INITIAL_ACCOUNT_STATE,
    INITIAL_ALARM_STATE,
    INITIAL_BUDGET_STATE,
    INITIAL_CATEGORY_STATE,
    INITIAL_DEBT_STATE,
    INITIAL_DISPOSABLE_STATE,
    INITIAL_EXPENSE_STATE
} from "../test_helper/initial_state";

Enzyme.configure({adapter: new Adapter()});

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe('Expenses', () => {
    it('renders ExpenseOverview as expected', () => {
        const wrapper = shallow(
            <ExpenseOverview/>,
            {
                context: {
                    store: mockStore({
                        expense: INITIAL_EXPENSE_STATE,
                        budget: INITIAL_BUDGET_STATE,
                        debt: INITIAL_DEBT_STATE,
                        disposable: INITIAL_DISPOSABLE_STATE,
                        category: INITIAL_CATEGORY_STATE,
                        alarm: INITIAL_ALARM_STATE,
                        account: INITIAL_ACCOUNT_STATE
                    })
                }
            },
        );
        expect(wrapper.dive()).toMatchSnapshot();
    });

    it('renders ExpenseOverview, filled total expenses, as expected', () => {
        let newExpenseState = INITIAL_EXPENSE_STATE;
        newExpenseState.totalExpenses = 50;
        const wrapper = shallow(
            <ExpenseOverview/>,
            {
                context: {
                    store: mockStore({
                        expense: newExpenseState,
                        budget: INITIAL_BUDGET_STATE,
                        debt: INITIAL_DEBT_STATE,
                        disposable: INITIAL_DISPOSABLE_STATE,
                        category: INITIAL_CATEGORY_STATE,
                        alarm: INITIAL_ALARM_STATE,
                        account: INITIAL_ACCOUNT_STATE
                    })
                }
            },
        );
        expect(wrapper.dive()).toMatchSnapshot();
    });
});
