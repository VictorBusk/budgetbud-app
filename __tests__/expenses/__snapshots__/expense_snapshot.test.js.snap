// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expenses renders ExpenseOverview as expected 1`] = `
<Styled(Container)
  style={
    Object {
      "backgroundColor": "#ffffff",
    }
  }
>
  <AppHeader
    headerText="Expense overview"
    infoText="Here you see an overview of your spending compared to your budget. Press the bell icon on any expense item to get a notification if your spendings exceed the expense goal."
    onLeftButtonPress={[Function]}
  />
  <Styled(Container)>
    <Styled(Container)>
      <Styled(ViewNB)
        style={
          Object {
            "flex": 0.82,
          }
        }
      >
        <FlatList
          data={Array []}
          disableVirtualization={false}
          horizontal={false}
          initialNumToRender={10}
          keyExtractor={[Function]}
          maxToRenderPerBatch={10}
          numColumns={1}
          onEndReachedThreshold={2}
          onRefresh={[Function]}
          refreshing={false}
          renderItem={[Function]}
          scrollEventThrottle={50}
          updateCellsBatchingPeriod={50}
          windowSize={21}
        />
        <Separator />
      </Styled(ViewNB)>
      <Styled(ViewNB)
        style={
          Object {
            "flex": 0.18,
          }
        }
      >
        <Styled(ViewNB)
          style={
            Array [
              Object {
                "flex": 1,
                "justifyContent": "space-between",
                "marginBottom": 0,
              },
              Object {
                "paddingVertical": "5%",
              },
            ]
          }
        >
          <Styled(ViewNB)
            style={
              Array [
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingLeft": 15,
                  "paddingRight": 18,
                },
                Object {
                  "flex": 1,
                },
              ]
            }
          >
            <Styled(Text)
              style={
                Array [
                  Object {
                    "alignSelf": "flex-start",
                    "fontSize": 16,
                    "marginLeft": 5,
                  },
                  Object {
                    "color": "#003755",
                  },
                ]
              }
            >
              Total debt per month
            </Styled(Text)>
            <Styled(Text)
              style={
                Array [
                  Object {
                    "alignSelf": "flex-start",
                    "fontSize": 16,
                    "marginLeft": 5,
                  },
                  Object {
                    "color": "#003755",
                  },
                ]
              }
            >
              0
               
              kr
            </Styled(Text)>
          </Styled(ViewNB)>
          <Styled(ViewNB)
            style={
              Array [
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingLeft": 15,
                  "paddingRight": 18,
                },
                Object {
                  "flex": 1,
                },
              ]
            }
          >
            <Styled(Text)
              style={
                Array [
                  Object {
                    "alignSelf": "flex-start",
                    "fontSize": 16,
                    "marginLeft": 5,
                  },
                  Object {
                    "color": "#003755",
                  },
                ]
              }
            >
              Total expenses
            </Styled(Text)>
            <Styled(Text)
              style={
                Array [
                  Object {
                    "alignSelf": "flex-start",
                    "fontSize": 16,
                    "marginLeft": 5,
                  },
                  Object {
                    "color": "#003755",
                  },
                ]
              }
            >
              0
               
              kr
            </Styled(Text)>
          </Styled(ViewNB)>
          <Styled(ViewNB)
            style={
              Array [
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingLeft": 15,
                  "paddingRight": 18,
                },
                Object {
                  "flex": 1,
                },
              ]
            }
          >
            <Styled(Text)
              style={
                Array [
                  Object {
                    "alignSelf": "flex-start",
                    "fontSize": 16,
                    "marginLeft": 5,
                  },
                  Object {
                    "color": "#003755",
                  },
                ]
              }
            >
              Disposable
            </Styled(Text)>
            <Styled(Text)
              style={
                Array [
                  Object {
                    "alignSelf": "flex-start",
                    "fontSize": 16,
                    "marginLeft": 5,
                  },
                  Object {
                    "color": "#003755",
                  },
                ]
              }
            >
              0
               
              kr
            </Styled(Text)>
          </Styled(ViewNB)>
        </Styled(ViewNB)>
      </Styled(ViewNB)>
    </Styled(Container)>
  </Styled(Container)>
</Styled(Container)>
`;

exports[`Expenses renders ExpenseOverview, filled total expenses, as expected 1`] = `
<Styled(Container)
  style={
    Object {
      "backgroundColor": "#ffffff",
    }
  }
>
  <AppHeader
    headerText="Expense overview"
    infoText="Here you see an overview of your spending compared to your budget. Press the bell icon on any expense item to get a notification if your spendings exceed the expense goal."
    onLeftButtonPress={[Function]}
  />
  <Styled(Container)>
    <Styled(Container)>
      <Styled(ViewNB)
        style={
          Object {
            "flex": 0.82,
          }
        }
      >
        <FlatList
          data={Array []}
          disableVirtualization={false}
          horizontal={false}
          initialNumToRender={10}
          keyExtractor={[Function]}
          maxToRenderPerBatch={10}
          numColumns={1}
          onEndReachedThreshold={2}
          onRefresh={[Function]}
          refreshing={false}
          renderItem={[Function]}
          scrollEventThrottle={50}
          updateCellsBatchingPeriod={50}
          windowSize={21}
        />
        <Separator />
      </Styled(ViewNB)>
      <Styled(ViewNB)
        style={
          Object {
            "flex": 0.18,
          }
        }
      >
        <Styled(ViewNB)
          style={
            Array [
              Object {
                "flex": 1,
                "justifyContent": "space-between",
                "marginBottom": 0,
              },
              Object {
                "paddingVertical": "5%",
              },
            ]
          }
        >
          <Styled(ViewNB)
            style={
              Array [
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingLeft": 15,
                  "paddingRight": 18,
                },
                Object {
                  "flex": 1,
                },
              ]
            }
          >
            <Styled(Text)
              style={
                Array [
                  Object {
                    "alignSelf": "flex-start",
                    "fontSize": 16,
                    "marginLeft": 5,
                  },
                  Object {
                    "color": "#003755",
                  },
                ]
              }
            >
              Total debt per month
            </Styled(Text)>
            <Styled(Text)
              style={
                Array [
                  Object {
                    "alignSelf": "flex-start",
                    "fontSize": 16,
                    "marginLeft": 5,
                  },
                  Object {
                    "color": "#003755",
                  },
                ]
              }
            >
              0
               
              kr
            </Styled(Text)>
          </Styled(ViewNB)>
          <Styled(ViewNB)
            style={
              Array [
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingLeft": 15,
                  "paddingRight": 18,
                },
                Object {
                  "flex": 1,
                },
              ]
            }
          >
            <Styled(Text)
              style={
                Array [
                  Object {
                    "alignSelf": "flex-start",
                    "fontSize": 16,
                    "marginLeft": 5,
                  },
                  Object {
                    "color": "#003755",
                  },
                ]
              }
            >
              Total expenses
            </Styled(Text)>
            <Styled(Text)
              style={
                Array [
                  Object {
                    "alignSelf": "flex-start",
                    "fontSize": 16,
                    "marginLeft": 5,
                  },
                  Object {
                    "color": "#003755",
                  },
                ]
              }
            >
              50
               
              kr
            </Styled(Text)>
          </Styled(ViewNB)>
          <Styled(ViewNB)
            style={
              Array [
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingLeft": 15,
                  "paddingRight": 18,
                },
                Object {
                  "flex": 1,
                },
              ]
            }
          >
            <Styled(Text)
              style={
                Array [
                  Object {
                    "alignSelf": "flex-start",
                    "fontSize": 16,
                    "marginLeft": 5,
                  },
                  Object {
                    "color": "#003755",
                  },
                ]
              }
            >
              Disposable
            </Styled(Text)>
            <Styled(Text)
              style={
                Array [
                  Object {
                    "alignSelf": "flex-start",
                    "fontSize": 16,
                    "marginLeft": 5,
                  },
                  Object {
                    "color": "#003755",
                  },
                ]
              }
            >
              0
               
              kr
            </Styled(Text)>
          </Styled(ViewNB)>
        </Styled(ViewNB)>
      </Styled(ViewNB)>
    </Styled(Container)>
  </Styled(Container)>
</Styled(Container)>
`;
